File: backend/apps/user/urls.py
from django.urls import path

from .views import (
    AdminToUserView,
    AuthorizedUserListView,
    BlockUserView,
    ProfileAddPhotoView,
    ProfileListView,
    ProfileUpdateView,
    UnBlockUserView,
    UserDestroyView,
    UserListCreateView,
    UserToAdminView
)

urlpatterns = [
    path('', UserListCreateView.as_view(), name='user_list_create'),
    path('/delete', UserDestroyView.as_view(), name='user_destroy'),
    path('/online', AuthorizedUserListView.as_view(), name='user_online'),
    path('/<int:pk>/block', BlockUserView.as_view(), name='user_block'),
    path('/<int:pk>/unblock', UnBlockUserView.as_view(), name='user_unblock'),
    path('/<int:pk>/to_admin', UserToAdminView.as_view(), name='user_to_admin'),
    path('/<int:pk>/to_user', AdminToUserView.as_view(), name='admin_to_user'),
    path('/profiles', ProfileListView.as_view(), name='profile_list'),
    path('/profiles/patch', ProfileUpdateView.as_view(), name='profile_update'),
    path('/profiles/add_photo', ProfileAddPhotoView.as_view(), name='profile_photos'),
]

File: frontend/src/constants/urls.ts
const baseURL = import.meta.env.VITE_API_URL;

const auth = '/auth';
const posts = '/posts';
const users = '/users'
const profiles = '/profiles'

const urls = {
    auth: {
        login: auth,
        refresh: auth + '/refresh',
        activate: auth + '/activate',
        socket: auth + '/socket'
    },
    posts: {
        list_create: posts,
        retrieve_update_destroy: (id: string) => {
            return posts + '/' + id
        }
    },
    users: {
        list_create: users,
        list_online: users + '/online',
        destroy: users + '/delete',
        block: (id: string) => {
            return users + `/${id}/block`
        },
        unblock: (id: string) => {
            return users + `/${id}/unblock`
        },
        user_to_admin: (id: string) => {
            return users + `/${id}/to_admin`
        },
        admin_to_user: (id: string) => {
            return users + `/${id}/to_user`
        },
        profiles: {
            list: users + profiles,
            update: users + profiles + '/patch',
            add_photo: (id: string) => {
                return users + profiles + `/${id}/photos`
            }
        }
    }
}

export {
    baseURL,
    urls
}
